src/client/Microsoft.Identity.Client/MsalManagedIdentityException.cs

```diff
@@ -11,6 +11,7 @@ namespace Microsoft.Identity.Client
     /// This exception class is for exceptions generated from Managed Identity sources.
     /// For more details, see https://aka.ms/msal-net-managed-identity
     /// </summary>
+    [Obsolete("MsalManagedIdentityException is deprecated. Use MsalServiceException instead.")]
```

Comment:
Add the flag that breaks the build and set it to true - this can be a hard deprecation.